addons/l10n_ro_efactura/models/ciusro_document.py

```diff
@@ -0,0 +1,205 @@
+    state = fields.Selection(
+        selection=[
+            ('invoice_sending', 'Sending'),
+            ('invoice_sending_failed', 'Error'),
+            ('invoice_sent', 'Sent'),
+        ],
+        string='E-Factura Status',
+        required=True,
+    )
+    datetime = fields.Datetime(default=fields.Datetime.now, required=True)
+    attachment_id = fields.Many2one(comodel_name='ir.attachment')
+    message = fields.Char()
+    key_loading = fields.Char()
+    key_download = fields.Char()
+    key_signature = fields.Char()
+    key_certificate = fields.Char()
+    need_fetch_button = fields.Boolean(compute='_compute_need_fetch_button')
+
+    @api.model
+    def _create_document_invoice_sending(self, invoice, key_loading: str):
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending',
+            'key_loading': key_loading,
+        })
+
+    @api.model
+    def _create_document_invoice_sending_failed(self, invoice, message: str):
+        self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sending_failed',
+            'message': message,
+        })
+
+    @api.model
+    def _create_document_invoice_sent(self, invoice, result: dict):
+        document = self.env['l10n_ro_edi.document'].create({
+            'invoice_id': invoice.id,
+            'state': 'invoice_sent',
+            'key_signature': result['key_signature'],
+            'key_certificate': result['key_certificate'],
+        })
+        document.attachment_id = self.env['ir.attachment'].create({
+            'name': invoice._l10n_ro_edi_get_attachment_file_name(),
+            'raw': result['attachment_raw'],
+            'res_model': self._name,
+            'res_id': document.id,
+            'type': 'binary',
+            'mimetype': 'application/xml',
+        })
+
+    def unlink(self):
+        """ Make sure any created attachments are also deleted """
+        self.attachment_id.unlink()
+        return super().unlink()
+
+    @api.model
+    def _make_efactura_request(self, company, endpoint, method, params, data=None, session=None):
+        """ Make an API request to E-Factura and handle the response and return a `result` dictionary
+        :param company: `res.company` object containing l10n_ro_edi_access_token
+        :param endpoint: 'upload' (for sending) |
+                         'stareMesaj' (for fetching status) |
+                         'descarcare' (for downloading answer)
+        :param method: 'post' (for 'upload') |
+                       'get' (for 'stareMesaj'|'descarcare')
+        :param params: Dictionary of query parameters
+        :param data: xml data for 'upload' request
+        :return: dict of {'error': <str>|<bytes>} or {'content': <response.content>} from E-Factura
+        """
+        url = f"https://api.anaf.ro/test/FCTEL/rest/{endpoint}"
```

Comment:
missing:
        if not company.l10n_ro_edi_test_env:
            url = f"https://api.anaf.ro/prod/FCTEL/rest/{endpoint}"